python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -131,14 +131,33 @@ def dist_inference_one_layer(layer_id, g, dataloader, target_ntypes, layer, get_
     -------
     dict of Tensors : the inferenced tensors.
     """
+    len_dataloader = max_num_batch = len(list(dataloader))
```

Comment:
You should also take a look of node_mini_batch_gnn_predict and edge_mini_batch_gnn_predict

Reply:
They should have the same problem as here.

Reply:
The fixes for node_mini_batch_gnn_predict  are in node_gnn.py

Reply:
Since we are not using edge features, do we need to modify edge_gnn.py?

Reply:
edge_gnn also access node features

Reply:
You are right! Fixed.